src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -728,162 +760,215 @@ DatabaseShardImp::onChildrenStopped()
+
+            ProgressDisplay progress(
+                numLedgers / stepSize,
+                "Object",
+                (boost::format("Shard %1%: Verifying SQLite objects...") %
+                 shardIndex)
+                    .str());
 
-            // Verify SQLite ledgers are in the node store
+            bool valid{true};
+            for (std::uint32_t n = firstSeq; n <= lastSeq; n += stepSize)
             {
-                auto const firstSeq{firstLedgerSeq(shardIndex)};
-                auto const lastSeq{
-                    std::max(firstSeq, lastLedgerSeq(shardIndex))};
-                auto const numLedgers{
-                    shardIndex == earliestShardIndex() ? lastSeq - firstSeq + 1
-                                                       : ledgersPerShard_};
-                auto ledgerHashes{getHashesByIndex(firstSeq, lastSeq, app_)};
-                if (ledgerHashes.size() != numLedgers)
-                    continue;
-
-                bool valid{true};
-                for (std::uint32_t n = firstSeq; n <= lastSeq; n += 256)
+                if (!source.fetchNodeObject(ledgerHashes[n].first, n))
                 {
-                    if (!source.fetchNodeObject(ledgerHashes[n].first, n))
-                    {
-                        JLOG(j_.warn()) << "SQLite ledger sequence " << n
-                                        << " mismatches node store";
-                        valid = false;
-                        break;
-                    }
+                    JLOG(j_.warn()) << "SQLite ledger sequence " << n
+                                    << " mismatches node store";
+                    valid = false;
+                    break;
                 }
-                if (!valid)
-                    continue;
+                ++progress;
             }
+            if (!valid)
+                continue;
+        }
 
-            auto const path =
-                needsHistoricalPath ? chooseHistoricalPath(lock) : dir_;
+        auto const path = needsHistoricalPath
+            ? chooseHistoricalPath(std::lock_guard(mutex_))
+            : dir_;
 
-            // Create the new shard
-            auto shard{
-                std::make_unique<Shard>(app_, *this, shardIndex, path, j_)};
-            if (!shard->init(scheduler_, *ctx_))
-                continue;
+        // Create the new shard
+        auto shard{std::make_shared<Shard>(app_, *this, shardIndex, path, j_)};
+        if (!shard->init(scheduler_, *ctx_))
+            continue;
 
-            // Create a marker file to signify an import in progress
-            auto const shardDir{path / std::to_string(shardIndex)};
-            auto const markerFile{shardDir / importMarker_};
+        // Sweep the shard cache to conserve memory
+        ThreadWrapper shardSweeper(
+            [sweepInterval, wptr = std::weak_ptr<Shard>(shard)] {
+                if (auto shard = wptr.lock(); shard)
+                    shard->sweep();
```

Comment:
The shard sweep does nothing. `shardSweeper` can be eliminated.

Reply:
It's been removed